client/widgets/VideoWidget.cpp

```diff
@@ -91,6 +113,34 @@ void VideoWidgetBase::stopAudio()
 	}
 }
 
+std::string VideoWidgetBase::getSubTitleLine(double timestamp)
+{
+	if(srtContent.empty())
+		return "";
+	
+    std::regex exp("^\\s*(\\d+:\\d+:\\d+,\\d+)[^\\S\\n]+-->[^\\S\\n]+(\\d+:\\d+:\\d+,\\d+)((?:\\n(?!\\d+:\\d+:\\d+,\\d+\\b|\\n+\\d+$).*)*)", std::regex::multiline);
```

Comment:
wow that looks terrible... A parser would be better, simply istream might even work. Please also add a comment with line format, then we'll see how to improve it.

Reply:
Yes. Also a non noticed todo. Was the fastest way to get it basically working. Just copy an existing regex.

I've only published it as draft to show that I'm currently working on it. 😅